nixos/modules/services/monitoring/vnstat.nix

```diff
@@ -6,21 +6,19 @@ let
   cfg = config.services.vnstat;
 in {
   options.services.vnstat = {
-    enable = mkOption {
-      type = types.bool;
-      default = false;
-      description = ''
-        Whether to enable update of network usage statistics via vnstatd.
-      '';
-    };
+    enable = mkEnableOption "update of network usage statistics via vnstatd";
   };
 
   config = mkIf cfg.enable {
-    users.users.vnstatd = {
-      isSystemUser = true;
-      description = "vnstat daemon user";
-      home = "/var/lib/vnstat";
-      createHome = true;
+
+    environment.systemPackages = [ pkgs.vnstat ];
+
+    users = {
+      groups.vnstatd.members = [ "vnstatd" ];
```

Comment:
Usually this is written as
```
users.users.vnstatd.extraGroups = [ "vnstatd" ];
users.groups.vnstatd = {};
```

I had assumed the two to be equivalent, but given your findings maybe `userss.groups.vnstatd.members` doesn't set the **primary** group but adds as a secondary? I'm sorry, I don't have the time to dig into this at the moment...

Additionally, you should add `systemd.services.vnstat.serviceConfig.Group = "vnstatd";`.

Reply:
jup, that works
(the `extraGroups` line doesn't seem to affect the effects i'm seeing, but i guess it can't hurt to be verbose)

Reply:
Sorry what am I saying... I meant:

```
users.users.vnstatd.groups = "vnstatd";
```

Reply:
ok, that makes more sense :D